src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -840,84 +829,21 @@ static void Main()
-  IL_0000:  ldc.i4.1
+  // Code size       31 (0x1f)
+  .maxstack  3
+  .locals init (int V_0) //y
+  IL_0000:  ldc.i4.4
   IL_0001:  conv.i8
-  IL_0002:  ldc.i4.1
-  IL_0003:  conv.i8
-  IL_0004:  ldc.i4.1
-  IL_0005:  conv.i8
-  IL_0006:  ldc.i4.1
-  IL_0007:  conv.i8
-  IL_0008:  ldc.i4.1
-  IL_0009:  conv.i8
-  IL_000a:  ldc.i4.1
-  IL_000b:  conv.i8
-  IL_000c:  ldc.i4.1
-  IL_000d:  conv.i8
-  IL_000e:  ldc.i4.1
-  IL_000f:  conv.i8
-  IL_0010:  ldc.i4.4
-  IL_0011:  conv.i8
-  IL_0012:  ldc.i4.2
-  IL_0013:  newobj     ""System.ValueTuple<long, long, int>..ctor(long, long, int)""
-  IL_0018:  newobj     ""System.ValueTuple<long, long, long, long, long, long, long, (long, long, int)>..ctor(long, long, long, long, long, long, long, (long, long, int))""
-  IL_001d:  dup
-  IL_001e:  ldfld      ""long System.ValueTuple<long, long, long, long, long, long, long, (long, long, int)>.Item1""
-  IL_0023:  stloc.1
-  IL_0024:  dup
-  IL_0025:  ldfld      ""long System.ValueTuple<long, long, long, long, long, long, long, (long, long, int)>.Item2""
-  IL_002a:  stloc.2
-  IL_002b:  dup
-  IL_002c:  ldfld      ""long System.ValueTuple<long, long, long, long, long, long, long, (long, long, int)>.Item3""
-  IL_0031:  stloc.3
-  IL_0032:  dup
-  IL_0033:  ldfld      ""long System.ValueTuple<long, long, long, long, long, long, long, (long, long, int)>.Item4""
-  IL_0038:  stloc.s    V_4
-  IL_003a:  dup
-  IL_003b:  ldfld      ""long System.ValueTuple<long, long, long, long, long, long, long, (long, long, int)>.Item5""
-  IL_0040:  stloc.s    V_5
-  IL_0042:  dup
-  IL_0043:  ldfld      ""long System.ValueTuple<long, long, long, long, long, long, long, (long, long, int)>.Item6""
-  IL_0048:  stloc.s    V_6
-  IL_004a:  dup
-  IL_004b:  ldfld      ""long System.ValueTuple<long, long, long, long, long, long, long, (long, long, int)>.Item7""
-  IL_0050:  stloc.s    V_7
-  IL_0052:  dup
-  IL_0053:  ldfld      ""(long, long, int) System.ValueTuple<long, long, long, long, long, long, long, (long, long, int)>.Rest""
-  IL_0058:  ldfld      ""long System.ValueTuple<long, long, int>.Item1""
-  IL_005d:  stloc.s    V_8
-  IL_005f:  dup
-  IL_0060:  ldfld      ""(long, long, int) System.ValueTuple<long, long, long, long, long, long, long, (long, long, int)>.Rest""
-  IL_0065:  ldfld      ""long System.ValueTuple<long, long, int>.Item2""
-  IL_006a:  stloc.s    V_9
-  IL_006c:  ldfld      ""(long, long, int) System.ValueTuple<long, long, long, long, long, long, long, (long, long, int)>.Rest""
-  IL_0071:  ldfld      ""int System.ValueTuple<long, long, int>.Item3""
-  IL_0076:  stloc.s    V_10
-  IL_0078:  ldloc.s    V_9
-  IL_007a:  ldloc.s    V_10
-  IL_007c:  stloc.0
-  IL_007d:  box        ""long""
-  IL_0082:  ldstr      "" ""
-  IL_0087:  ldloc.0
-  IL_0088:  box        ""int""
-  IL_008d:  call       ""string string.Concat(object, object, object)""
-  IL_0092:  call       ""void System.Console.WriteLine(string)""
-  IL_0097:  ret
-}
-");
+  IL_0002:  ldc.i4.2
```

Comment:
Yeah, this is definitely more optimal :-)

Now that we do not bounce the values off an intermediate tuple variable, optimizer can figure what is going on, elides dead assignments and then kills unused temps